File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/model/Email.java
package com.eager.questioncloud.core.domain.verification.model;

import com.eager.questioncloud.core.domain.verification.template.EmailVerificationTemplate;
import com.eager.questioncloud.core.domain.verification.template.EmailVerificationTemplateCreator;
import lombok.Getter;

@Getter
public class Email {
    private String to;
    private String subject;
    private String content;

    public Email(String to, String subject, String content) {
        this.to = to;
        this.subject = subject;
        this.content = content;
    }

    public static Email of(EmailVerification emailVerification) {
        EmailVerificationTemplate template = EmailVerificationTemplateCreator.getTemplate(
            emailVerification.getEmailVerificationType(),
            emailVerification.getToken());

        return new Email(emailVerification.getEmail(), template.getTitle(), template.getContent());
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/model/Email.kt
package com.eager.questioncloud.core.domain.verification.model

import com.eager.questioncloud.core.domain.verification.template.EmailVerificationTemplateCreator

class Email(
    val to: String,
    val subject: String?,
    val content: String?
) {
    companion object {
        @JvmStatic
        fun of(emailVerification: EmailVerification): Email {
            val template = EmailVerificationTemplateCreator.getTemplate(
                emailVerification.emailVerificationType,
                emailVerification.token
            )

            return Email(emailVerification.email, template.title, template.content)
        }
    }
}
